src/api.ts

```diff
@@ -0,0 +1,187 @@
+import fs from 'fs';
+import path from 'path';
+import glob from 'glob';
+
+import { generateHtml } from './html';
+import { exploreBundle, UNMAPPED_KEY } from './explore';
+import { AppError, getErrorMessage } from './app-error';
+import {
+  BundlesAndFileTokens,
+  ExploreOptions,
+  ExploreResult,
+  Bundle,
+  ExploreErrorResult,
+  ExploreBundleResult,
+  ErrorCode,
+} from './index';
+
+/**
+ * Analyze bundle(s)
+ */
+export async function explore(
+  bundlesAndFileTokens: BundlesAndFileTokens,
+  options: ExploreOptions = {}
+): Promise<ExploreResult> {
+  const bundles: Bundle[] = [];
+  const fileTokens: string[] = [];
+
+  bundlesAndFileTokens = Array.isArray(bundlesAndFileTokens)
+    ? bundlesAndFileTokens
+    : [bundlesAndFileTokens];
+
+  if (bundlesAndFileTokens.length === 0) {
+    throw new AppError(ErrorCode.NoBundles);
+  }
+
+  // Separate bundles from file tokens
+  bundlesAndFileTokens.forEach(item => {
+    if (typeof item === 'string') {
+      fileTokens.push(item);
+    } else {
+      bundles.push(item);
+    }
+  });
+
+  // Get bundles from file tokens
+  bundles.push(...getBundles(fileTokens));
+
+  return Promise.all(
```

Comment:
Since you're using `async` anyway, any reason not to rewrite this with `await`?

```ts
const results = await Promise.all(...);
const result = getExploreResult(results, options);
// ...
```

Reply:
No, I will use `await`

Reply:
Done